RecoMuon/DetLayers/src/MuonGEMDetLayerGeometryBuilder.cc

```diff
@@ -22,103 +22,81 @@ MuonGEMDetLayerGeometryBuilder::~MuonGEMDetLayerGeometryBuilder() {}
 // Builds the forward (first) and backward (second) layers
 // Builds etaPartitions (for rechits)
 pair<vector<DetLayer*>, vector<DetLayer*> > MuonGEMDetLayerGeometryBuilder::buildEndcapLayers(const GEMGeometry& geo) {
-  vector<DetLayer*> result[2];
+  vector<DetLayer*> endcapLayers[2];
 
-  for (int endcap = -1; endcap <= 1; endcap += 2) {
-    int iendcap = (endcap == 1) ? 0 : 1;  // +1: forward, -1: backward
+  for (auto st : geo.stations()) {
+    for (int layer = GEMDetId::minLayerId + 1; layer <= GEMDetId::maxLayerId0; ++layer) {
+      if (st->station() != GEMDetId::minStationId0 && layer > GEMDetId::maxLayerId)
+        break;
 
-    for (int station = GEMDetId::minStationId; station <= GEMDetId::maxStationId; ++station) {
-      for (int layer = GEMDetId::minLayerId + 1; layer <= GEMDetId::maxLayerId; ++layer) {
-        vector<int> rolls, rings, chambers;
-        for (int ring = GEMDetId::minRingId; ring <= GEMDetId::maxRingId; ++ring)
-          rings.push_back(ring);
-        for (int chamber = GEMDetId::minChamberId + 1; chamber <= GEMDetId::maxChamberId; chamber++)
-          chambers.push_back(chamber);
-        for (int roll = GEMDetId::minRollId + 1; roll <= GEMDetId::maxRollId; ++roll)
-          rolls.push_back(roll);
+      ForwardDetLayer* fowardLayer = nullptr;
```

Comment:
Since you are introducing it, you can fix the typo: `fowardLayer` -> `forwardLayer`